core/src/main/java/com/google/errorprone/bugpatterns/collectionincompatibletype/CollectionUndefinedEquality.java

```diff
@@ -44,6 +48,16 @@
     severity = WARNING)
 public final class CollectionUndefinedEquality extends BugChecker
     implements MethodInvocationTreeMatcher, MemberReferenceTreeMatcher {
+  // NOTE: this is a bit crude; methods like `containsValue` are still likely to be subject to
+  // equality constraints on the value type.
+  private static final Matcher<ExpressionTree> TYPES_NOT_DEPENDING_ON_OBJECT_EQUALITY =
+      anyMethod()
+          .onClass(
+              isDescendantOfAny(
+                  ImmutableList.of(
+                      "java.util.IdentityHashMap",
+                      "java.util.IdentityHashSet",
+                      "java.util.TreeMap")));
```

Comment:
> I thought of TreeMap and IdentityHash{Set,Map}. Do any more come to mind?

I think here `java.util.TreeMap` can be replaced with `java.util.SortedMap`; the same caveats w.r.t. `#compareTo` vs. `#equals` are [documented](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/util/SortedMap.html) at the interface level.